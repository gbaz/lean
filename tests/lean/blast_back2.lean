constant r : nat → Prop
constant s : nat → Prop
constant p : nat → Prop

definition q (a : nat) := p a

lemma rq₁ [intro] [priority 20] : ∀ a, r a → q a :=
sorry

lemma rq₂ [intro] [priority 10] : ∀ a, s a → q a :=
sorry

attribute q [reducible]

definition lemma1 (a : nat) : r a → s a → p a :=
by blast

print lemma1

attribute rq₂ [intro] [priority 30]

definition lemma2 (a : nat) : r a → s a → p a :=
by blast

print lemma2
