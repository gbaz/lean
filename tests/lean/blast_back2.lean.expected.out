definition lemma1 : ∀ (a : ℕ), r a → s a → p a :=
λ (a : ℕ) (H.1 : r a) (H.2 : s a), rq₁ a H.1
definition lemma2 : ∀ (a : ℕ), r a → s a → p a :=
λ (a : ℕ) (H.1 : r a), rq₂ a
